Good Coding Practices/src/js/scripts.js

```diff
@@ -1,44 +1,112 @@
 // piece object
-const piece = (function() {
+
+const piece = (function () {
   let el = null;
-  const init = function(el) {
+
+  const init = function (el) {
     this.el = el;
   };
-  const moveDelta = function(dx, dy) {
+
+  const moveDelta = function (dx, dy) {
     const pos = this.el.getBoundingClientRect();
-    this.el.style.left = `${pos.left + dx}px`;
-    this.el.style.top = `${pos.top + dy}px`;
+    const position = [pos.top, pos.left];
+    if (dx === 0) {
+      this.el.style.top = `${movePos(position[0], dy,window.innerHeight - 100)}px`;
+    }
+    else
+      this.el.style.left = `${movePos(position[1], dx,window.innerWidth - 100)}px`;
+  }
+
+  const movePos = function(pos, px, screenSize) {   
+    let positionSide=0;
+    if ((pos + px < screenSize) && (pos + px > 0))
+      positionSide = pos + px;
+    else if (pos + px >= screenSize)
+      positionSide = screenSize;
+    else if (pos.top + px <= 0)
+      positionSide = 0;
+    return positionSide;
+  }
+
+  const pieceColor = function (temp) {
+    let color = "";
+    if (temp <= 10) {
+      color = "blue";
+    }
+    else if (temp > 10 && temp < 21)
+      color = "Green";
+    else if (temp > 20 && temp < 31)
+      color = "Yellow";
+    else if (temp > 30)
+      color = "Red";
+    this.el.style.background = color;
+    this.el.style.borderColor = color;
+  }
+
+  const weatherApi = function () {
+    fetch('http://api.apixu.com/v1/current.json?key=dda6e762ae4f41efb7e173552192204&q=tel%20aviv')

```

Comment:
This knowledge does not belong in `piece`
(API URL, result data structure, API key!)